core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,77 +55,147 @@ object ConsumerGroupCommand {
+          case None =>
+            printError(s"The consumer group '$groupId' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              state match {
+                case Some("Dead") =>
+                  printError(s"Consumer group '$groupId' does not exist.")
+                case Some("Empty") =>
+                  printError(s"Consumer group '$groupId' has no active members.")
+                case Some(_) =>
+                  printError(s"Consumer group '$groupId' is rebalancing.")
+                case None =>
+                  // the control should never reach here
+                  throw new KafkaException("Expected a valid consumer group state, but none found.")
+              }
+            else
+              printAssignment(assignments)
+        }
+      }
       else if (opts.options.has(opts.deleteOpt)) {
         consumerGroupService match {
-          case service: ZkConsumerGroupService => service.delete()
-          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
+          case service: ZkConsumerGroupService => service.deleteGroups()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService.")
         }
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        printError(s"Executing consumer group command failed due to ${e.getMessage}", Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
+  def printError(msg: String, e: Option[Throwable] = None): Unit = {
+    println(s"Error: $msg")
+    e.foreach(debug("Exception in consumer group command", _))
+  }
+
+  def printAssignment(groupAssignment: Seq[PartitionAssignmentState]): Unit = {
+    require(groupAssignment.nonEmpty)
+    println("Consumer group: " + groupAssignment.head.group)
+    groupAssignment.head.coordinator match {
+      case Some(node) => println(s"Consumer group coordinator broker id: ${node.id}")
+      case default =>
+    }
+
+    print("\n%-30s %-10s %-15s %-15s %-10s %-50s".format("TOPIC", "PARTITION", "CURRENT-OFFSET", "LOG-END-OFFSET", "LAG", "CONSUMER-ID"))
+    if (groupAssignment.head.coordinator.nonEmpty)
+      print("%-30s %s".format("HOST", "CLIENT-ID"))
+    println()
+
+    groupAssignment.foreach { memberAssignment =>
+      print("%-30s %-10s %-15s %-15s %-10s %-50s".format(
+        memberAssignment.topic.getOrElse(""), memberAssignment.partition.getOrElse(""),
+        memberAssignment.offset.getOrElse(""), memberAssignment.logEndOffset.getOrElse(""), memberAssignment.lag.getOrElse(""),
+        memberAssignment.consumerId.getOrElse("none")))
+      if (groupAssignment.head.coordinator.nonEmpty)
+        print("%-30s %s".format(memberAssignment.host.getOrElse(""), memberAssignment.clientId.getOrElse("")))
+      println()
```

Comment:
BTW, I'm thinking printing "-" instead of "" when data is not available or does not apply would look better and more readable. Compare
```
TOPIC                          PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG        CONSUMER-ID
test1                          0          0               0               0          consumer-1-7913cdd0-e177-409f-86fa-3fcf92ddd926
                                                                                     consumer-1-b2f82fe9-6c1a-4928-8e6d-ccfab0906788
```
with
```
TOPIC                          PARTITION  CURRENT-OFFSET  LOG-END-OFFSET  LAG        CONSUMER-ID
test1                          0          0               0               0          consumer-1-7913cdd0-e177-409f-86fa-3fcf92ddd926
-                              -          -               -               -          consumer-1-b2f82fe9-6c1a-4928-8e6d-ccfab0906788
```